sql/src/main/java/io/druid/sql/calcite/rule/SelectRules.java

```diff
@@ -152,10 +150,8 @@ public void onMatch(RelOptRuleCall call)
         return;
       }
 
-      // Only push in sorts that can be used by the Select query.
-      final List<OrderByColumnSpec> orderBys = limitSpec.getColumns();
-      if (orderBys.isEmpty() ||
-          (orderBys.size() == 1 && orderBys.get(0).getDimension().equals(Column.TIME_COLUMN_NAME))) {
+      // Scan query can handle limiting but not sorting, so avoid applying this rule if there is a sort.
+      if (limitSpec.getColumns().isEmpty()) {
```

Comment:
This patch would have made order by time impossible. I guess I could add the "select" query back, just so this kind of query is possible. It will take some more time.